src/Illuminate/Foundation/Application.php

```diff
@@ -552,6 +558,15 @@ protected function markAsRegistered($provider)
 		$this->serviceProviders[] = $provider;
 
 		$this->loadedProviders[$class] = true;
+		
+		$aliases = class_parents($class) + class_implements($class) + array($class);
```

Comment:
My own micro-benchmarks have shown that using a temporary variable has almost no performance impact, I think I've been testing that since 5.2 and it's still the case in 5.5.  The difference between the two varies and is well within the margin for error.